### ISSUE TITLE
Dialyzer error on 1.3.4

### ISSUE THREAD

**Issue by RobinClowers** [appsignal/appsignal-elixir#270]
I just upgraded from 1.2.3 to 1.3.4, and now I'm seeing a dialyzer error: `lib/error_tracker.ex:26: Function track/4 has no local return`. Here is my track function and a helper function:

```elixir
  def track(error, message, stack, metadata) do
    IO.warn inspect metadata
    Appsignal.send_error(error, message, stack, to_meta(metadata))
    :ok
  end

  def to_meta(data) when is_map(data) do
    for {key, value} <- data, into: %{}, do: {key, inspect(value)}
  end
  def to_meta(data), do: %{arguments: inspect(data)}

```

Pretty simple, and I didn't change anything in those functions when I updated. I suspect the internal workings of the `send_error` function changed in a way dialyzer didn't like, but I didn't see anything obvious when I looked at the diff on github. Any ideas?

**Comment by tombruijn** [comment]
Hi @RobinClowers ,

We added another param in #260. That might be causing the problem? Not sure about what that dialyzer error means. We still have an issue open for adding dialyzer to this repo in #165, but haven't gotten around to it.

We'll try to have a look soon-ish. Is it breaking your app, right now or can you work around/ignore the error?

**Comment by RobinClowers** [comment]
@tombruijn Thanks for getting back to me. You can configure dialyzer to ignore errors in specific functions, so I did that to work around the issue. I saw the added param, I don't think that should be the issue since it's optional. I couldn't make much sense of the error either, it basically means "your function typespec says it always returns some value, but there are cases where it can raise an exception and never return". The odd thing is that I tried changing the type def to include `no_return` and I still get the error.

**Comment by tombruijn** [comment]
Will be fixed in #271
